### PR TITLE
Improve clarity of SQSMessageContext

### PR DESCRIPTION
**Pull request by jessemyers**
The opaque/context abstraction is still hard to follow and the pubsub usage herein
has some unclear assumptions, but we can simplify a little without much pain

### CODE REVIEW

File: microcosm_pubsub/dispatcher.py [hunk 1]
```diff
@@ -50,25 +48,25 @@ def handle_message(self, message, bound_handlers):
...
+            extra.update(
                 handler=titleize(handler.__class__.__name__),
                 uri=content.get("uri"),
-            ))
+            )
 
-            if self.enable_ttl:
-                ttl = content.get("X-Request-TTL", self.initial_ttl) - 1
+            if self.sqs_message_context.enable_ttl:
+                ttl = content.get("X-Request-TTL", self.sqs_message_context.initial_ttl) - 1
```
Comment (netama) [inline microcosm_pubsub/dispatcher.py]: use `TTL_KEY` instead?

File: microcosm_pubsub/dispatcher.py [hunk 2]
```diff
@@ -50,25 +48,25 @@ def handle_message(self, message, bound_handlers):
...
-            extra.update(dict(
+            extra.update(
                 handler=titleize(handler.__class__.__name__),
                 uri=content.get("uri"),
-            ))
+            )
 
-            if self.enable_ttl:
-                ttl = content.get("X-Request-TTL", self.initial_ttl) - 1
+            if self.sqs_message_context.enable_ttl:
```
[the remark this answers is not in the document]
Comment (netama) [reply microcosm_pubsub/dispatcher.py]: yes, it looks like it might even reduce the TTL by two instead of one?

[the remark this answers is not in the document]
Comment (pelegdvir) [reply microcosm_pubsub/dispatcher.py]: @jessemyers - you can remove it:
I've moved this code out of `dispatcher.py` in https://github.com/globality-corp/microcosm-pubsub/pull/115/files. master branch is fine.
But for some reason - merging it back to develop somehow brought it back and I didn't notice
It's also the reason you had to open https://github.com/globality-corp/microcosm-pubsub/pull/123

File: microcosm_pubsub/dispatcher.py (hunk 1 above)
Comment (pelegdvir) [reply microcosm_pubsub/dispatcher.py]: Just remove this code

File: microcosm_pubsub/dispatcher.py [hunk 3]
```diff
@@ -50,25 +48,25 @@ def handle_message(self, message, bound_handlers):
...
             return False
 
-        with self.opaque.initialize(self.sqs_message_context, content, message_id=message_id):
-            try:
-                handler = self.sqs_message_handler_registry.find(media_type, bound_handlers)
-            except KeyError:
-                # no handlers
-                self.logger.debug("Skipping message with no registered handler: {}".format(media_type))
-                return False
+        try:
```
[the remark this answers is not in the document]
Comment (pelegdvir) [reply microcosm_pubsub/dispatcher.py]: I think you can
